### PR TITLE
Add integration-tests for overlayfs' mount-syscall-interception feature [skip ci]

### PR DESCRIPTION
**Pull request by rodnymolina**
Signed-off-by: Rodny Molina <rmolina@nestybox.com>

### CODE REVIEW

File: tests/syscall/mount-overlayfs.bats [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+ovs_path="/var/lib/docker/overlay2"
+
+function teardown() {
+  sysbox_log_check
+}
+
+# Verify basic overlayfs mount instruction.
+@test "mount overlayfs basic" {
+
+  local lower_path="$ovs_path"/root/lower
```
Comment (ctalledo) [inline tests/syscall/mount-overlayfs.bats:22]: I would also add files and subdir under lower and upper:

lower/l1
lower/ldir/l2
upper/u1
upper/udir/u2

And then verify that the merged dir has the correct hierarchy: 

l1
ldir/l2
u1
udir/u2

That way we know that ovfs it's doing its thing.

File: tests/syscall/mount-overlayfs.bats [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+  # Add new dir.
+  docker exec "$syscont" bash -c "mkdir $merge_path/new_dir"
+  [ "$status" -eq 0 ]
+
+  # Add new file.
+  docker exec "$syscont" bash -c "touch $merge_path/new_file"
+  [ "$status" -eq 0 ]
+
+  # Confirm new dir/file is present.
+  docker exec "$syscont" bash -c "stat $merge_path/new_dir $merge_path/new_file"
```
Comment (ctalledo) [inline tests/syscall/mount-overlayfs.bats:91]: Let's also verify that the new file is present in the upper dir.

File: tests/syscall/mount-overlayfs.bats [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+  docker exec "$syscont" bash -c "umount $merge_path"
+  [ "$status" -eq 0 ]
+
+  verify_syscont_overlay_umnt $syscont $merge_path
+
+  docker_stop "$syscont"
+}
+
+# Mix of various overlayfs mounts and umounts operations from chroot'ed contexts,
+# creation of whiteouts, etc -- fairly decent cocktail of ovs mount instructions.
```
Comment (ctalledo) [inline tests/syscall/mount-overlayfs.bats:179]: ovs -> ovfs

File: tests/syscall/mount-overlayfs.bats [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+# Verify overlayfs mounts with relative paths.
+@test "mount overlayfs relative paths" {
+
+  local lower_path="$ovs_path"/root/lower
+  local upper_path="$ovs_path"/root/upper
+  local work_path="$ovs_path"/root/work
+  local merge_path="$ovs_path"/root/merge
+
+  local rel_lower_dir=root/lower
```
Comment (ctalledo) [inline tests/syscall/mount-overlayfs.bats:203]: how about "./lower" (use the "." to force path resolution).

File: tests/syscall/mount-overlayfs.bats (hunk 1 above)
Comment (rodnymolina) [reply tests/syscall/mount-overlayfs.bats:22]: This is a very basic test to simply verify if mount completes. The other testcases go much deeper into what you are suggesting.

File: tests/syscall/mount-overlayfs.bats (hunk 2 above)
Comment (rodnymolina) [reply tests/syscall/mount-overlayfs.bats:91]: Wondering why are you suggesting this. If new file is not present in the upper dir, then we would be dealing with a kernel bug, not a sysbox one.

File: tests/syscall/mount-overlayfs.bats (hunk 3 above)
Comment (rodnymolina) [reply tests/syscall/mount-overlayfs.bats:179]: sure.

File: tests/syscall/mount-overlayfs.bats (hunk 2 above)
Comment (ctalledo) [reply tests/syscall/mount-overlayfs.bats:91]: yes you are right, but it's a sanity check to make sure we are not messing up anything in the way sysbox processes the ovfs mount.

File: tests/syscall/mount-overlayfs.bats (hunk 1 above)
Comment (ctalledo) [reply tests/syscall/mount-overlayfs.bats:22]: I didn't see any other test that does what I suggested above (but maybe I missed it).

File: tests/syscall/mount-overlayfs.bats (hunk 2 above)
Comment (rodnymolina) [reply tests/syscall/mount-overlayfs.bats:91]: Oh i see what you mean: add files, folders, and files-within-folders. Sure will add this to the next testcase.

File: tests/syscall/mount-overlayfs.bats (hunk 1 above)
Comment (rodnymolina) [reply tests/syscall/mount-overlayfs.bats:22]: Oh i see what you mean: add files, folders, AND files-within-folders. Sure, will add this extra level to the 'addition' testcase.

File: tests/syscall/mount-overlayfs.bats (hunk 2 above)
Comment (rodnymolina) [reply tests/syscall/mount-overlayfs.bats:93]: i did it anyways.

File: tests/syscall/mount-overlayfs.bats (hunk 4 above)
Comment (rodnymolina) [reply tests/syscall/mount-overlayfs.bats:209]: Don't see the need for this.

**Review by ctalledo — APPROVED** [review]

**Pull request merged**